Script to run manuscript analyses

sjspielman opened the pull request:
Closes #1595 

This PR adds `scripts/run-manuscript-analyses.sh` as well as associated documentation. 

I made some minor clarifying updates to the `analyses/README.md` file as well and fixed a typo. Please let me know if the added sentence here makes sense.


When reviewing, also see this comment, since I have two lines in there for running the telomerase and we'll need to pick one: https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/1595#issuecomment-1206495621

jaclyn-taroni commented:
@jashapiro can I get your opinion on this as well? I don't remember why we deviated from the plan originally in #1455:

> Create a fourth script, called something like `scripts/run-analysis-modules.sh`. This script will run all analysis modules that either i) were not run for subtyping, ii) were partially run for subtyping but now need to be fully run (e.g. GSVA), or iii) were run for sybtyping and need to be run again with the subtyped data.

What was added in #1543 was only analyses that fit those criteria ☝🏻  _and_ were required for making figures. #1593/#1594 was adding a 5th script that was other analyses in the manuscript fit those criteria but were not required for making figures.

I can see the point of including everything like what we have here, but the script being added here doesn't actually include everything because subtyping is entirely absent. I don't think we want to include subtyping, so it makes more sense to me to go back to what was originally stated in #1455 and add the following criterion: iv) needs to be re-run so files are available for figure creation (e.g., SNV callers). 

I would like to hear other opinions, though.

jashapiro commented:
> What was added in #1543 was only analyses that fit those criteria ☝🏻 _and_ were required for making figures. #1593/#1594 was adding a 5th script that was other analyses in the manuscript fit those criteria but were not required for making figures.
> 
> I can see the point of including everything like what we have here, but the script being added here doesn't actually include everything because subtyping is entirely absent. I don't think we want to include subtyping, so it makes more sense to me to go back to what was originally stated in #1455 and add the following criterion: iv) needs to be re-run so files are available for figure creation (e.g., SNV callers).

To clarify, is the difference here that  #1593/#1594 would have created two scripts and this PR combines those into a single script? So @jaclyn-taroni your preference would be to reopen #1594?

jaclyn-taroni commented:
My preference, which is not 100% solidified, is to have this script run modules that fit the following criteria:

* Non-deprecated modules that are used in the manuscript in some manner
* Were not run for data generation _unless_ the underlying `scratch` files are required for figures
* Were not run for subtyping or need to be re-run now with the disease label information in the release histologies file

instead of "running everything" included in the manuscript, because it isn't running everything – subtyping is missing and it should stay that way in my opinion. My argument is, in part, that running everything but subtyping might lull someone into a sense that they are running all the relevant modules, and that keeping things limited to what is necessary for the manuscript post-release is more straightforward.

jashapiro commented:
> My preference, which is not 100% solidified, is to have this script run modules that fit the following criteria:
> 
> * Non-deprecated modules that are used in the manuscript in some manner
> * Were not run for data generation _unless_ the underlying `scratch` files are required for figures
> * Were not run for subtyping or need to be re-run now with the disease label information in the release histologies file
> 
> instead of "running everything" included in the manuscript, because it isn't running everything – subtyping is missing and it should stay that way in my opinion. My argument is, in part, that running everything but subtyping might lull someone into a sense that they are running all the relevant modules, and that keeping things limited to what is necessary for the manuscript post-release is more straightforward.

Looking through in a bit more detail, it seems that there are modules run here (`independent-samples` was the one I noted) that _were_ run for data release generation in `generate-analysis-files-for-release.sh`, and I would agree those should _not_ be rerun here unless required  to populate `scratch`.

But I am fine with combining all scripts that need to be run for figures _or other ms results_, as was my general interpretation of replacing #1593/#1594 with this.

Are we saying the same thing?

jaclyn-taroni commented:
I believe we are saying the same thing, yea.

sjspielman commented:
Just going to note that I (for now!) included `independent-samples` here because the code would _not_ have previously been run with the _subtyped_ metadata file - https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/independent-samples#generating-sample-lists
It's been my sense that anything run with `generate-analysis-files-for-release.sh` would _not_ have used the release's subtyping information, so relevant modules that could consume the subtyped metadata should be re-run. I do feel pretty strongly that whatever results are in the analyses modules at "release" (preprint/pub/send repo to Zenodo) time should be up-to-date, and it seems to me if we don't re-run analyses that consume the metadata file, then we have outdated analyses. Does this make sense/am I missing something?

jashapiro commented:
> Just going to note that I (for now!) included `independent-samples` here because the code would _not_ have previously been run with the _subtyped_ metadata file

True, but I don't think it uses or writes that data at all, so there should be no need to rerun it.

sjspielman commented:
I've updated the docs here to be more precise about which scripts are run, and I removed modules that I think should be removed:
- `collapse-rnaseq`
- `copy_number_consensus_call`
- `hotspot-detection`
- `independent-samples`

I _think_ `fusion_filtering` and or `fusion-summary` can also be removed, but I wasn't certain? Feedback on those modules as well as the ones I did remove would be great, thanks!

jashapiro commented on figures/README.md:
```diff
@@ -78,6 +78,7 @@ RUN_LOCAL=1 bash ../scripts/run-analyses-for-figures.sh
 _⚠️ This requires 64GB of RAM to run successfully! You can set `RUN_LOCAL=1` for local testing that skips the RAM-intensive steps._
 
 The script [`generate-figures.sh`](./generate-figures.sh) runs all scripts in [`figures/scripts/`](./scripts/) to generate manuscript-ready figure panels, as well as copies certain panels from analyses that are meant for the manuscript.
+**Note that it is ASSUMED that all results, including certain files in `scratch/`, will have been generated with `../scripts/run-manuscript-analyses.sh` before running this script.**
```
Should we add a check to the `generate-figures.sh` script? Or is the concern here that there will be old versions of the files hanging around? If so, I would make sure to say this directly.

If we want to get fancy, we could check in the figure script that the relevant files in `scratch/`  are not too old and print a warning if they are older than, say, a week.

sjspielman replied:
>  If so, I would make sure to say this directly.
👍 

> If we want to get fancy, we could check in the figure script that the relevant files in scratch/ are not too old and print a warning if they are older than, say, a week.

I think this is a good way to check!

sjspielman replied:
> though I will admit I did go through
Do you mean "did not"? Seems like doing it is not an admission :) 

I am pretty certain that the order is good, based on what results feed into other modules.

jashapiro reviewed (commented):
This looks generally fine to me, though I will admit I did go through with a fine comb to be sure all of the analyses are there and in order. I did have a comment about checking requirements for later scripts though...

jaclyn-taroni reviewed (approved):
I agree with what came out. I'm returning my thoughts on the fusion modules and approving with the expectation that those will be implemented.

sjspielman commented:
I removed `fusion-summary`, but kept `fusion_filtering`, and I added some `scratch/` checks into `figures/generate-figures.sh` along with some updated documentation.

jashapiro commented on figures/generate-figures.sh:
```diff
@@ -43,6 +43,39 @@ mkdir -p pdfs/supp/figs6/panels
+snv_tcga_db_check=${scratch_dir}/snv-callers/tcga_snv_db.sqlite
+snv_pbta_scratch_days=$((($(date +%s) - $(date +%s -r "$snv_pbta_db_check")) / 86400))  # https://unix.stackexchange.com/questions/102691/get-age-of-given-file
+snv_tcga_scratch_days=$((($(date +%s) - $(date +%s -r "$snv_tcga_db_check")) / 86400))  # https://unix.stackexchange.com/questions/102691/get-age-of-given-file
+if [ ! -f $snv_pbta_db_check ] || [ ! -f $snv_tcga_db_check ]; then
+  echo "The 'snv-callers' module scratch databases do not exist. Please re-run the 'snv-callers' module first, or run the script 'scripts/run-manuscript-analyses.sh'."
+  exit 0
+fi
```
This check should probably be before the date calculations, or I expect they will fail. 
Also, since this is an error, `exit 1`

sjspielman replied:
done!

jashapiro commented on figures/generate-figures.sh:
```diff
@@ -43,6 +43,40 @@ mkdir -p pdfs/supp/figs6/panels
+  exit 1
+fi
+
+snv_pbta_scratch_days=$((($(date +%s) - $(date +%s -r "$snv_pbta_db_check")) / 86400))  # https://unix.stackexchange.com/questions/102691/get-age-of-given-file
+snv_tcga_scratch_days=$((($(date +%s) - $(date +%s -r "$snv_tcga_db_check")) / 86400))  # https://unix.stackexchange.com/questions/102691/get-age-of-given-file
```
Wow that is sadly ugly.

sjspielman replied:
This was the _least_ ugly thing I found in stackoverflow/stackexchange land, which may not be very surprising.

jashapiro reviewed (approved):
LGTM
